packages/block-editor/src/components/link-control/style.scss

```diff
@@ -166,10 +166,6 @@ $block-editor-link-control-number-of-actions: 1;
 		text-overflow: ellipsis;
 	}
 
-	.block-editor-link-control__search-item-title mark {
-		color: $gray-900;
-	}
-
```

Comment:
A few lines below this part we set the color to `$black` so this rule was always overridden and never applied.